services/configuration/src/schema.rs

```diff
@@ -254,10 +314,64 @@ mod tests {
         let new_merkle_root = {
             let mut fork = db.fork();
             let mut index: ProofListIndex<_, MaybeVote> = ProofListIndex::new("index", &mut fork);
-            index.set(2, MaybeVote::some(vote.clone()));
+            index.set(2, MaybeVote::from(vote.clone()));
             index.set(2, MaybeVote::none());
             index.merkle_root()
         };
         assert_eq!(merkle_root, new_merkle_root);
     }
+
+    #[test]
+    fn test_serialization_of_voting_decision() {
+        const VALIDATORS: usize = 5;
+
+        let (pubkey, key) = crypto::gen_keypair();
+        let vote = Vote::new(&pubkey, &Hash::new([1; 32]), &key);
+        let vote_against = VoteAgainst::new(&pubkey, &Hash::new([1; 32]), &key);
+        assert_eq!(
+            vote.clone().into_bytes(),
+            VotingDecision::Yea(vote.clone()).into_bytes()
+        );
+        assert_eq!(vote.hash(), VotingDecision::Yea(vote.clone()).hash());
+        assert_eq!(
+            vote_against.clone().into_bytes(),
+            VotingDecision::Nay(vote_against.clone()).into_bytes()
+        );
+        assert_eq!(
+            vote_against.hash(),
+            VotingDecision::Nay(vote_against.clone()).hash()
+        );
+
+        let db = MemoryDB::new();
+        let mut fork = db.fork();
+        let merkle_root = {
+            let mut index: ProofListIndex<_, VotingDecision> =
```

Comment:
I don't quite understand the reasoning to test `ProofListIndex<_, VotingDecision>`, given that there are no such indexes in the schema (but `ProofListIndex<_, MaybeVote>` are). In general, this part of the test in the current form may be redundant: unlike the previous test, which tests compatibility between 2 different implementations of the schema, this one actually doesn't.

It still may make sense to test compatibility by comparing `ProofListIndex<_, Vote>` (which is equivalent to the previous `MaybeVote` implementation if some elements are set to `NO_VOTE`) and `ProofListIndex<_, MaybeVote>` (i.e., the new implementation). It should be possible to read from `ProofListIndex<_, MaybeVote>`, which was written as `ProofListIndex<_, Vote>`, and this conversion should not change the `merkle_root` of the index.

Reply:
You don't need to push `VoteAgainst` into the index - `VoteAgainst` was absent before this PR. The idea is to check that functionality that *was* present (i.e., `Vote` / `NO_VOTE`) hasn't been broken by this PR.

Reply:
If so, that is already covered by test_serialization_of_maybe_vote(). I think ProofListIndex test can be removed.